### ISSUE TITLE
Wrong execution order in afterStep/afterScenario

### ISSUE THREAD

**Issue by woolter** [webdriverio/wdio-cucumber-framework#151]
With this configuration
```
beforeScenario: function(scenario) {
    console.log(scenario.name);
  },
```

```
afterStep: function(stepResult) {
    console.log('Status:', stepResult.status);
    if (stepResult.status === 'failed') {
      browser.saveScreenshot(
        './errorShots/' +
          stepResult.feature +
          '_' +
          stepResult.scenario +
          '_' +
          stepResult.text +
          '.png'
      );
      console.log('Screenshot created');
    }
  },
```

```
afterScenario: function(scenario) {
    console.log(scenario.name);
  },
```

I hope to obtain this:
```
GOOGLE
Status: passed
Status: passed
GOOGLE
YANDEX
Status: passed
Status: failed
Screenshot created
Status: skipped
YANDEX
```
But sometimes I obtain this:
with image created
```
GOOGLE
Status: passed
Status: passed
GOOGLE
YANDEX
Status: passed
Status: failed
Status: skipped
YANDEX
Screenshot created
```
And sometimes this:
without create image
```
GOOGLE
Status: passed
Status: passed
GOOGLE
YANDEX
Status: passed
Status: failed
Status: skipped
YANDEX
```

**Comment by NaveenThiyagarajan** [comment]
@christian-bromann This is the issue i was talking about in ticket https://github.com/webdriverio/webdriverio/issues/3074
The issue is seen in the latest `wdio-cucumber-framework`.

This issue is not in the version `wdio-cucumber-framework: 1.0.3`. I have a project and confirmed.

I have created a sample git project for you to reproduce the isuse:
https://github.com/NaveenThiyagarajan/wdio_cuke_hookIssue.git

Thanks.
Naveen

**Comment by BorisOsipov** [comment]
@NaveenThiyagarajan see my comment in https://github.com/webdriverio/webdriverio/issues/3068#issuecomment-444011064

it will help you.

**Comment by NaveenThiyagarajan** [comment]
Thanks @BorisOsipov I tried using the cucumber default hooks

```
After(() => {
    if(Status.FAILED === 'failed'){
        browser.screenshot().value;
        console.log('Screenshot taken');
    }
});
```
... But the allure-reporter was acting up weird with that .. :( it was showing skipped tests as `passed`.

**Comment by BorisOsipov** [comment]
>... But the allure-reporter was acting up weird with that .. :( it was showing skipped tests as passed.

I haven't seen such issue in allure repo...

**Comment by NaveenThiyagarajan** [comment]
> > ... But the allure-reporter was acting up weird with that .. :( it was showing skipped tests as passed.
> 
> I haven't seen such issue in allure repo...

Ok .. I got it.. It fixed once i moved to `0.8.3` allure-reporter. Thank you.. good for now.
